Add Fauna integration (#2425)
DataDog/integrations-extras: fauna/README.md

# Agent Check: Fauna

## Overview

This integration allows Fauna users to import their database query logs into Datadog. Users are able to configure it for specific databases or for entire region groups. When enabled all requests made to fauna matching the database or region group configuration will show up in Datadog logs under the fauna service.

> To see what fields are included, see the [Fauna Query Log Record Format][1].

Reviewer: The overview should include: 
	1	What does your product do in 1-2 sentences?
	2	What will customers get from this integration (the value of the integration & why it's helpful)
	3	What specific data your integration monitors and the value?

## Setup

### Installation

> Use the 'Connect Accounts' button above to connect your Fauna and Datadog account. This will allow Fauna to send logs to your Datadog account and will take you through a brief configuration process.

Reviewer: This should include each step for set up the integration including each OAuth flow step

### Configuration

> Once your accounts are connected, you can visit the [Fauna Integrations Page][2] to modify the integration configuration.

Reviewer: What is being modified?

3. You will be directed to the Fauna Create Integration page, from here, select the region group(s) or databases that you would like to stream logs into Datadog for and click create.
4. When directed to Datadog, click Authorize to authorize Fauna to create an API key for your account that will be used to send the database query logs.

Once this flow is complete it will redirect you to the [Fauna Integrations Page][2] showing your active integration.
Within the next 10 minutes, any queries issued for the configured region group(s) or databases should now show up in the 
Datadog Log Explorer under the 'fauna' service.

### Configuration

Fauna allows configuring the following on your Datadog integration:
1. Which region groups that database query logs are sent for.
   1. If a region group is enabled for the integration, query logs are sent for every database in the enabled region group.
2. Which databases that query logs are sent for.
3. The integration state, Active or Paused. When Paused no logs will be sent.

Visit the [Fauna Integrations Page][2] to update any of the configuration or to remove your integration.

> It can take up to 10 minutes to see your updates take effect in the Datadog Log Explorer.

Reviewer 2: ```suggestion
It can take up to ten minutes to see your updates take effect in the Datadog Log Explorer.
```

### Validation

> Once configured correctly, you should begin seeing your queries showing up in the Datadog Log Explorer under the 'fauna' service.

Reviewer 2: ```suggestion
Once configured correctly, you should see your queries start to appear in the Datadog Log Explorer under the 'fauna' service.
```

### Uninstallation
 - Visit the [Fauna Integrations Page][2] and remove the Datadog integration.
 - Once this integration has been uninstalled, any previous authorizations are revoked.

>  - Additionally, ensure that all API keys associated with this integration have been disabled by searching for the integration name on the API Keys page.

Reviewer 2: ```suggestion
 - Additionally, ensure that all API keys associated with this integration have been disabled by searching for the integration name on the [API Keys page][5].
```

## Data Collected

### Logs

> Fauna will include logs for every query for the databases configured in the integration. 
> If the integration is configured with a region group, then query logs for all databases within the region group are included.

Reviewer 2: ```suggestion
Fauna includes logs for every query within the databases configured in the integration. 
If the integration is configured with a region group, then query logs for all databases within the region group are included.
```

## Troubleshooting

> Need help? Contact [Fauna support](mailto:support@fauna.com).

Reviewer 2: ```suggestion
Need help? Contact [Fauna support][6].
```

[1]: https://docs.fauna.com/fauna/current/tools/query_log/reference/log_reference

> [2]: https://dashboard.fauna.com/resources/integrations

Reviewer 2: ```suggestion
[2]: https://dashboard.fauna.com/resources/integrations
[3]: https://app.datadoghq.com/integrations/fauna
[4]: https://docs.datadoghq.com/logs/explorer/
[5]: https://app.datadoghq.com/organization-settings/api-keys
[6]: mailto:support@fauna.com
```

1. Fauna Cost Measures
   1. Read Ops
   2. Write Ops
   3. Compute Ops
2. Query Time
3. Query Size Measures
   1. Request and response size
4. Query shape
   1. The full request body with Personal Identifiable Data removed.
5. Query response code

For the full list of fields available in the logs, see the [Fauna Query Log Record Format][1].

## Setup

### Installation

1. Click the 'Connect Accounts' button on the [Fauna integration tile][3] to start the OAuth flow and connect your Fauna and Datadog accounts.
2. Once directed to Fauna, login with the account you'd like to connect. You can skip this step if you're already logged into your Fauna account.
3. You are redirected to the Fauna Create Integration page, where you can select the region group(s) or databases that you would like to stream logs into Datadog for. Then, click **Create**.
4. When directed to Datadog, click **Authorize** to authorize Fauna to create an API key for your account that will be used to send the database query logs.

Once this flow is complete, you are redirected to the [Fauna Integrations Page][2], which shows your active integration.

> Within the next ten minutes, the [Fauna Overview Dashboard][7] will begin showing metrics for any queries issued for the configured region group(s) or databases.

Reviewer: Can you clarify that the metrics are logs? This could confuse users who are expecting metrics but only see logs. If you are planning to send in metrics, you will need to include a list of metrics in a [metadata](https://docs.datadoghq.com/developers/integrations/check_references/#metrics-metadata-file) file within the pr.

Author: updated! apologies for the confusion

The above issued queries should also start appearing in the [Datadog Log Explorer][4] under the 'fauna' service.

### Configuration

You can configure the following settings for the Fauna integration:

1. Which region groups database query logs are sent to.
   1. If a region group is enabled for the integration, query logs are sent for every database in the enabled region group.
2. Which databases query logs are sent from.
3. The integration state (Active or Paused). When Paused, logs are not sent.

Visit the [Fauna Integrations Page][2] to update any configuration settings or to remove your integration.

It can take up to ten minutes to see your updates take effect in the [Fauna Overview Dashboard][7] and [Datadog Log Explorer][4].

### Validation

> Once configured correctly, within ten minutes you should see metrics in the [Fauna Overview Dashboard][7] and queries in the [Datadog Log Explorer][4] under the 'fauna' service.

Reviewer: If you aren't sending metric in, please update this to say logs
